### PR TITLE
Add migration guide from WindowsAzure.ServiceBus for EventHubs

### LINKED ISSUE #17934: Add migration guide from WindowsAzure.ServiceBus to Azure.Messaging.EventHubs
We currently have a migration guide to move from Microsoft.Azure.EventHubs to Azure.Messaging.EventHubs. Since there are active users of the older WindowsAzure.ServiceBus package (used for Event Hubs) as well, we should have a migration guide from this package as well. We need to decide whether we want to update the current migration guide or create a new one.

Same goes for Microsoft.Azure.ServiceBus.EventProcessorHost

Related issue for Service Bus: #17933

### PR DESCRIPTION
**Pull request by ramya-rao-a**
This PR adds a migration guide for Event Hubs customers looking to move from the older .NET framework packages `WindowsAzure.ServiceBus` and `Microsoft.Azure.ServiceBus.EventProcessorHost` to the latest packages. Resolves #17934

This is meant to be a starting point and further PRs can add more details regarding features that were dropped.

Approach taken in this PR:
- Copy the [MigrationGuide](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/eventhub/Azure.Messaging.EventHubs/MigrationGuide.md) meant for Microsoft.Azure.EventHubs* packages
- Update package names
- Update class names and code snippets that was present for Microsoft.Azure.EventHubs* packages
- Removed the section on checkpoint migration. We can add this later if we see user interest

Interestingly most of the concept mostly exactly the same between the two older packages. 

Tip for reviewers: Compare & review just the diff between this file and [migration guide it was copied from](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/eventhub/Azure.Messaging.EventHubs/MigrationGuide.md). I copied rest of the fluff from what @jsquire had before :)

Also, am totally open for a better name for the file than `MigrationGuideFromDotNetFramework`.

### CODE REVIEW

File: sdk/eventhub/Azure.Messaging.EventHubs/MigrationGuideFromDotNetFramework.md [hunk 1]
```diff
@@ -0,0 +1,696 @@
...
+The `Azure.Messaging.EventHubs` family of packages also provide the ability to share in some of the cross-service improvements made to the Azure development experience, such as:
+
+-   Using the new `Azure.Identity` library to share a single authentication between clients
+-   A unified diagnostics pipeline offering a common view of the activities across each of the client libraries
+
+## General changes
+
+### Package and namespaces
+
+Package names and the namespace root for the Azure client libraries for .NET have changed. Each will follow the pattern `Azure.[Area].[Service]` where the legacy clients followed the pattern `WindowsAzure.[Service]` or `Microsoft.Azure.[Service]`. This provides a quick and accessible means to help understand, at a glance, whether you are using the current generation or legacy clients.
```
Comment (ramya-rao-a) [inline sdk/eventhub/Azure.Messaging.EventHubs/MigrationGuideFromDotNetFramework.md:46]: Should we mention that the older namespace used was `Microsoft.Messaging.something`?

File: sdk/eventhub/Azure.Messaging.EventHubs/MigrationGuideFromDotNetFramework.md [hunk 2]
```diff
@@ -0,0 +1,696 @@
...
+{
+    await producer.CloseAsync();
+}
+```
+
+#### Publishing events with a partition key
+
+In the `WindowsAzure.ServiceBus` library, to publish events and allow the Event Hubs service to automatically assign the partition, the `EventHubClient` was used as the publisher for a batch created with a partition key specified as an option.
+
+```C# Snippet:EventHubs_Migrate_T1_PublishWithAPartitionKey
```
Comment (jsquire) [inline sdk/eventhub/Azure.Messaging.EventHubs/MigrationGuideFromDotNetFramework.md:197]:
```suggestion
```C#
```

File: sdk/eventhub/Azure.Messaging.EventHubs/MigrationGuideFromDotNetFramework.md [hunk 3]
```diff
@@ -0,0 +1,696 @@
...
+{
+    await producer.CloseAsync();
+}
+```
+
+#### Publishing events to a specific partition
+
+In the `WindowsAzure.ServiceBuss` library, to publish events and allow the Event Hubs service to automatically assign the partition, a `EventHubSender` is created using the `EventHubClient` and then used as the publisher for a batch created with its default configuration.
+
+```C# Snippet:EventHubs_Migrate_T1_PublishToSpecificPartition
```
Comment (jsquire) [inline sdk/eventhub/Azure.Messaging.EventHubs/MigrationGuideFromDotNetFramework.md:266]:
```suggestion
```C#
```

File: sdk/eventhub/Azure.Messaging.EventHubs/MigrationGuideFromDotNetFramework.md [hunk 4]
```diff
@@ -0,0 +1,696 @@
...
+
+#### Reading events from all partitions
+
+In `Microsoft.Azure.ServiceBus.EventProcessorHost`, the `EventProcessorHost` provides a framework that allows you to concentrate on writing your business logic to process events while the processor holds responsibility for concerns such as resiliency, balancing work between multiple instances of your program, and supporting the creation of checkpoints to preserve state. Developers would have to create and register a concrete implementation of `IEventProcessor` to begin consuming events.
+
+```C# Snippet:EventHubs_Migrate_T1_SimpleEventProcessor
```
Comment (jsquire) [inline sdk/eventhub/Azure.Messaging.EventHubs/MigrationGuideFromDotNetFramework.md:347]:
```suggestion
```C#
```

File: sdk/eventhub/Azure.Messaging.EventHubs/MigrationGuideFromDotNetFramework.md [hunk 5]
```diff
@@ -0,0 +1,696 @@
...
+                $"Event received for partition: '{context.PartitionId}', " +
+                $"Data: '{data}'");
+        }
+
+        return Task.CompletedTask;
+    }
+}
+```
+
+```C# Snippet:EventHubs_Migrate_T1_BasicEventProcessorHost
```
Comment (jsquire) [inline sdk/eventhub/Azure.Messaging.EventHubs/MigrationGuideFromDotNetFramework.md:381]:
```suggestion
```C#
```

File: sdk/eventhub/Azure.Messaging.EventHubs/MigrationGuideFromDotNetFramework.md [hunk 6]
```diff
@@ -0,0 +1,696 @@
...
+   processor.ProcessEventAsync -= processEventHandler;
+   processor.ProcessErrorAsync -= processErrorHandler;
+}
+```
+
+#### Reading events from a single partition
+
+In the `WindowsAzure.ServiceBus` library, to read events from a partition, a `EventHubReceiver` is created using the `EventHubClient` and events are read in a pull-based manner.
+
+```C# Snippet:EventHubs_Migrate_T1_ReadFromSpecificPartition
```
Comment (jsquire) [inline sdk/eventhub/Azure.Messaging.EventHubs/MigrationGuideFromDotNetFramework.md:566]:
```suggestion
```C#

**Review by jsquire — COMMENTED** [review]
Thanks, @ramya-rao-a, for putting this together!    I left a few suggestions that were mostly around formatting to avoid snippet code gen from interfering.    I did not do a deep validation of the T0 code, so we'll probably want to ask a member of the Event Hubs team with more familiarity for their thoughts.

I do wonder if we should consider a different filename.   Maybe something like `MigrationGuide_WindowsAzureServiceBus.md` or similar to make it more obvious?

File: sdk/eventhub/Azure.Messaging.EventHubs/MigrationGuideFromDotNetFramework.md [hunk 7]
```diff
@@ -0,0 +1,696 @@
...
+    {
+        var eventData = new EventData(Encoding.UTF8.GetBytes($"Event #{ index }"));
+
+        if (!eventBatch.TryAdd(eventData))
+        {
+            throw new Exception($"The event at { index } could not be added");
+        }
+    }
+
+    await client.SendBatchAsync(eventBatch.ToEnumerable());
```
[the remark this answers is not in the document]
Comment (ramya-rao-a) [reply sdk/eventhub/Azure.Messaging.EventHubs/MigrationGuideFromDotNetFramework.md:154]: I referred the [call signature for EventHubclient.SendBatchAsync](https://docs.microsoft.com/en-us/dotnet/api/microsoft.servicebus.messaging.eventhubclient.sendbatchasync?view=azure-dotnet#microsoft-servicebus-messaging-eventhubclient-sendbatchasync(system-collections-generic-ienumerable((microsoft-servicebus-messaging-eventdata)))) and the note on [EventDataBatch.ToEnumerable](https://docs.microsoft.com/en-us/dotnet/api/microsoft.servicebus.messaging.eventdatabatch.toenumerable?view=azure-dotnet#microsoft-servicebus-messaging-eventdatabatch-toenumerable) saying that it was meant to be used in `EventHubClient.SendBatchAsync`

Couldnt find anything that would take the EventDataBatch directly....

@serkantkaraca Do you know of any method in WindowsAzure.ServiceBus that takes the EventDataBatch as input for sending?

[the remark this answers is not in the document]
Comment (serkantkaraca) [reply sdk/eventhub/Azure.Messaging.EventHubs/MigrationGuideFromDotNetFramework.md:154]: @ramya-rao-a here > https://docs.microsoft.com/en-us/dotnet/api/microsoft.servicebus.messaging.eventhubclient.sendasync?view=azure-dotnet#microsoft-servicebus-messaging-eventhubclient-sendasync(microsoft-servicebus-messaging-eventdata)

**Comment by ramya-rao-a** [comment]
@jsquire, File renamed and code snippets updated to pass the `verify generated code` step

File: sdk/eventhub/Azure.Messaging.EventHubs/MigrationGuide_WindowsAzureServiceBus.md [hunk 8]
```diff
@@ -0,0 +1,692 @@
...
+}
+finally
+{
+    await producer.CloseAsync();
+}
+```
+
+#### Publishing events to a specific partition
+
+In the `WindowsAzure.ServiceBuss` library, to publish events and allow the Event Hubs service to automatically assign the partition, a `EventHubSender` is created using the `EventHubClient` and then used as the publisher for a batch created with its default configuration.
```
Comment (JamesBirdsall) [inline sdk/eventhub/Azure.Messaging.EventHubs/MigrationGuide_WindowsAzureServiceBus.md:261]: ...to publish events to a specific, user-specified partition, an 'EventHubSender'...
This scenario is about NOT allowing the service to assign!

File: sdk/eventhub/Azure.Messaging.EventHubs/MigrationGuideFromDotNetFramework.md [hunk 9]
```diff
@@ -0,0 +1,696 @@
...
+var consumerGroupName = "<< NAME OF THE EVENT HUB CONSUMER GROUP >>";
+
+EventHubClient client = EventHubClient.CreateFromConnectionString(connectionString, eventHubName);
+EventHubConsumerGroup consumerGroup = ehClient.GetConsumerGroup(consumerGroupName);
+EventHubReceiver receiver = default;
+
+try
+{
+    string firstPartition = (await client.GetRuntimeInformationAsync()).PartitionIds.First();
+    receiver = consumerGroup.CreateReceiver(firstPartition, EventHubConsumerGroup.StartOfStream);
```
[the remark this answers is not in the document]
Comment (ramya-rao-a) [reply sdk/eventhub/Azure.Messaging.EventHubs/MigrationGuideFromDotNetFramework.md]: There is no ServiceBusClient in the old package. And the [EventHubClient](https://docs.microsoft.com/en-us/dotnet/api/microsoft.servicebus.messaging.eventhubclient?view=azure-dotnet) I found does not have a way to create the receiver. Are you thinking of the [MessageRecevier](https://docs.microsoft.com/en-us/dotnet/api/microsoft.servicebus.messaging.messagereceiver?view=azure-dotnet) ?

Or are you referring to [EventHubConsumerGroup.CreateReceiverAsync](https://docs.microsoft.com/en-us/dotnet/api/microsoft.servicebus.messaging.eventhubconsumergroup.createreceiverasync?view=azure-dotnet)?

File: sdk/eventhub/Azure.Messaging.EventHubs/MigrationGuideFromDotNetFramework.md [hunk 10]
```diff
@@ -0,0 +1,696 @@
...
+        Debug.WriteLine($"Partition: '{context.PartitionId}' was initialized.");
+        return Task.CompletedTask;
+    }
+
+    public Task ProcessEventsAsync(PartitionContext context, IEnumerable<EventData> messages)
+    {
+        foreach (var eventData in messages)
+        {
+            var data = Encoding.UTF8.GetString(
+                eventData.Body.Array,
```
[the remark this answers is not in the document]
Comment (ramya-rao-a) [reply sdk/eventhub/Azure.Messaging.EventHubs/MigrationGuideFromDotNetFramework.md]: Thanks @serkantkaraca! Fixed in https://github.com/Azure/azure-sdk-for-net/pull/26219/commits/6ded07152791aa1639896827c6f18da08f87395f. Also updated to print only the length of the body to match what the other code snippets are doing

File: sdk/eventhub/Azure.Messaging.EventHubs/MigrationGuideFromDotNetFramework.md [hunk 11]
```diff
@@ -0,0 +1,696 @@
...
+}
+finally
+{
+    await producer.CloseAsync();
+}
+```
+
+#### Publishing events to a specific partition
+
+In the `WindowsAzure.ServiceBuss` library, to publish events and allow the Event Hubs service to automatically assign the partition, a `EventHubSender` is created using the `EventHubClient` and then used as the publisher for a batch created with its default configuration.
```
[the remark this answers is not in the document]
Comment (ramya-rao-a) [reply sdk/eventhub/Azure.Messaging.EventHubs/MigrationGuideFromDotNetFramework.md]: Fixed in https://github.com/Azure/azure-sdk-for-net/pull/26219/commits/24b740e9aad0ca976bee6449885dfb0ffb426070

File: sdk/eventhub/Azure.Messaging.EventHubs/MigrationGuide_WindowsAzureServiceBus.md (hunk 8 above)
Comment (ramya-rao-a) [reply sdk/eventhub/Azure.Messaging.EventHubs/MigrationGuide_WindowsAzureServiceBus.md]: Thanks @JamesBirdsall!

Fixed both here and in the other migration guide: https://github.com/Azure/azure-sdk-for-net/pull/26219/commits/19f9ffee89054a2ebbd03da9da68ff72de36f0f1

File: sdk/eventhub/Azure.Messaging.EventHubs/MigrationGuideFromDotNetFramework.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (serkantkaraca) [reply sdk/eventhub/Azure.Messaging.EventHubs/MigrationGuideFromDotNetFramework.md]: Yes, I am referring to EventHubConsumerGroup.CreateReceiverAsync. CreateReceiver is an async operation in track-0.

[the remark this answers is not in the document]
Comment (ramya-rao-a) [reply sdk/eventhub/Azure.Messaging.EventHubs/MigrationGuideFromDotNetFramework.md]: Thanks @serkantkaraca 
Updated to use CreateReceiverAsync in https://github.com/Azure/azure-sdk-for-net/pull/26219/commits/3d56087df10dcd5bc8a16c44c6d1ad2bb010c747

**Review by serkantkaraca — APPROVED** [review]
Thx Ramya

**Pull request merged**